### PR TITLE
Updated readme, more tests, remove brackets, fraction input with decimals

### PR DESCRIPTION
**Pull request by flokixdev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,3 +154,124 @@ Before PR, please run complete code check via command:
...
+to create an extension on each layer and its sublayers.
+
+**Mathematical framework layers** ordered from the most concrete
+one to most abstract one:
+
+<table>
+    <tr>
+        <td>
+            <b>
+            <a href="https://github.com/cothema/math-php-api">
```
Comment (janbarasek) [inline README.md]: Why is this package first?

File: src/SmartNumber.php [hunk 2]
```diff
@@ -360,23 +358,24 @@ private function setValue(string $input): void
 		} catch (NumberFormatException $e) {
 		}
 
-		if (preg_match('/^(?<mantissa>-?\d*[.]?\d+)(e|E|^)(?<exponent>-?\d*[.]?\d+)$/', $input, $parseExponential)) {
```
Comment (janbarasek) [inline src/SmartNumber.php:363]: Where is support for numbers like `3E6` or `25.13e17`?

File: README.md (hunk 1 above)
Comment (flokixdev) [reply README.md]: Because it is the most concrete one (as said in the text). Please suggest a change.

File: src/SmartNumber.php (hunk 2 above)
Comment (janbarasek) [reply src/SmartNumber.php:363]: sorry.

Comment (flokixdev) [reply src/SmartNumber.php:363]: It is supported natively by brick math (BigNumber::of())

File: README.md (hunk 1 above)
Comment (flokixdev) [reply README.md]: Or should I change the order from the most abstract one to the most concrete one? I think that API package can be very useful, so it's there.

Comment (janbarasek) [reply README.md]: I think the Mathematicator should only use the packages it guarantees.

If it is a third-party package, it must be marked.

Comment (flokixdev) [reply README.md]: Ok, I'll move this one to the end and mark it as a third party package.

**Comment by janbarasek** [comment]
Thanks!

**Pull request merged**